### PR TITLE
Refactor issue 2

### PR DESCRIPTION
**Pull request by ragurney**

### CODE REVIEW

**Comment by ragurney** [comment]
I forgot to add updates from issue #3. Will change label back after changes are included in this PR.

File: db/migrate/20160826181115_change_images_timestamps_not_null.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ChangeImagesTimestampsNotNull < ActiveRecord::Migration[5.0]
+  def change
+    change_column_null(:images, :created_at, false, Time.zone.now)
```
Comment (bboe) [inline db/migrate/20160826181115_change_images_timestamps_not_null.rb]: Will this migration work if you already have data in the database?

File: db/schema.rb [hunk 2]
```diff
@@ -10,12 +10,37 @@
...
-ActiveRecord::Schema.define(version: 20160819175531) do
+ActiveRecord::Schema.define(version: 20160826181115) do
 
   create_table "images", force: :cascade do |t|
     t.string   "url",        null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "taggings", force: :cascade do |t|
```
Comment (bboe) [inline db/schema.rb]: Looks like this doesn't quite belong here. Please double check the changed files when opening a pull request to ensure everything looks as it should.

File: db/migrate/20160826181115_change_images_timestamps_not_null.rb (hunk 1 above)
Comment (ragurney) [inline db/migrate/20160826181115_change_images_timestamps_not_null.rb]: I think so. From what I was reading in the documentation, the optional fourth argument should replace any existing nulls in the database with its value. Is that right?

File: db/schema.rb (hunk 2 above)
Comment (ragurney) [inline db/schema.rb]: I've edited the commit to include only relevant changes. See e9d3443.

File: db/migrate/20160826181115_change_images_timestamps_not_null.rb (hunk 1 above)
Comment (bboe) [inline db/migrate/20160826181115_change_images_timestamps_not_null.rb]: I actually was not aware of the last argument. Awesome!

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -19,7 +19,7 @@ def create
 
   def show
     @image = Image.find(params['id'])
-  rescue ActiveRecord::RecordNotFound => _e
```
Comment (bboe) [inline app/controllers/images_controller.rb]: Is there a better name for this commit than "Refactor image controller"? In general "refactor" is pretty vague, and I don't think that necessarily applies to this commit.

File: test/flows/images_flow_test.rb [hunk 4]
```diff
@@ -30,6 +30,6 @@ class ImagesFlowTest < FlowTestCase
```
Comment (bboe) [inline test/flows/images_flow_test.rb]: We should avoid having commits pertaining to issues with git "Resolve merge conflict". In general, if there are merge conflicts, they should be resolved within whatever atomic commit makes sense. Perhaps this commit was accidentally renamed differently than expected.

**Comment by bboe** [comment]
Added a few additional comments.

File: test/flows/images_flow_test.rb (hunk 4 above)
Comment (ragurney) [inline test/flows/images_flow_test.rb]: I believe I accidentally committed the merge resolution rather than just do a git rebase --continue. I've squashed the offending commit into its original (4a32667).

File: app/controllers/images_controller.rb (hunk 3 above)
Comment (ragurney) [inline app/controllers/images_controller.rb]: 6c9764d

File: test/flows/images_flow_test.rb (hunk 4 above)
Comment (bboe) [inline test/flows/images_flow_test.rb]: I think this should still be in it's own commit. Since the title of this one is: Rename img to image

**Comment by bboe** [comment]
Almost there.

File: test/flows/images_flow_test.rb (hunk 4 above)
Comment (ragurney) [inline test/flows/images_flow_test.rb]: Ok just put it in its own commit: b6ebfa2.

**Comment by bboe** [comment]
👍
